lib/streamlit/runtime/state/query_params.py

```diff
@@ -91,8 +91,8 @@ def clear(self) -> None:
         self._query_params.clear()
         self._send_query_param_msg()
 
-    def to_dict(self) -> Dict[str, Union[List[str], str]]:
-        return self._query_params
+    def to_dict(self) -> Dict[str, str]:
+        return {key: self[key] for key in self._query_params}
```

Comment:
It'd be worth mentioning that writing `self[key]` here does the job of only getting the last query param if multiple are set. Otherwise this line is confusing since it's not clear why you're not just writing `return self._query_params` or `return self._query_params.copy()`